libs/log/tm_logger.go

```diff
@@ -52,6 +53,14 @@ func NewTMLoggerWithColorFn(w io.Writer, colorFn func(keyvals ...interface{}) te
 // Info logs a message at level Info.
 func (l *tmLogger) Info(msg string, keyvals ...interface{}) {
 	lWithLevel := kitlevel.Info(l.srcLogger)
+
+	// Encode all []byte types as base-16 (uppercase).
+	for i, val := range keyvals {
+		if b, ok := val.([]byte); ok {
+			keyvals[i] = bytes.ToUpper(b)
```

Comment:
bytes.ToUpper doesn't encode as hexademical (base16). in general stdlib provides `encoding/hex` module, it might not encode in upper-case. also it is a bit wasteful to do it here.

if value implements either `encoding.TextMarshaler` or `fmt.Stringer` it will be marshalled accordingly to the implementation after passing log level check

Reply:
Hi @dshulyak, thank you for the feedback! For my own knowledge, why might the encoding/hex module might not encode in uppercase?

Reply:
library is simply using lowercase [chars](https://github.com/golang/go/blob/master/src/encoding/hex/hex.go#L15) for encoding. i guess it is a common preference